src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/CrossTargetSubscriptionHostBase.cs

```diff
@@ -152,7 +152,11 @@ await subscriber.Value.InitializeSubscriberAsync(this, _activeConfiguredProjectS
 
         private async Task UpdateProjectContextAndSubscriptionsAsync()
         {
-            var previousProjectContext = _currentAggregateProjectContext;
+            var previousProjectContext = await ExecuteWithinLockAsync(() =>
+            {
+                return Task.FromResult(_currentAggregateProjectContext);
+            });
+
```

Comment:
yep, this is the only pace where we forgot to wrap access to _currentAggregateProjectContext in a lock.  If it is not locked, there could be race condition resulting in unnecessary context update and re-subscriptions from all subscribers, which would result in the whole round of new DT events.